osslab/src/app/ossdocker.py

```diff
@@ -0,0 +1,147 @@
+__author__ = 'junbo'
+
+import json
+
+import requests
+
+import log
+
+
+def convert(input):
+    if isinstance(input, dict):
+        return {convert(key):convert(value) for key,value in input.iteritems()}
+    elif isinstance(input, list):
+        return [convert(element) for element in input]
+    elif isinstance(input, unicode):
+        return input.encode('utf-8')
+    else:
+        return input
+
+
+def name_match(id, l):
+    for n in l:
+        if id in n:
+            return True
+
+    return False
+
+
+class OssDocker(object):
+    def __init__(self):
+        self.base_url = "http://localhost:4243"
+        self.headers = {'content-type': 'application/json'}
+
+    def containers_info(self):
+        try:
+            containers_url = self.base_url + "/containers/json"
+            req = requests.get(containers_url)
+            return convert(json.loads(req.content))
+        except Exception as e:
+            log.error(e)
+
+
+    def get_container(self, name):
+        containers = self.containers_info()
+        return next((c for c in containers if name in c["Names"] or '/' + name in c["Names"]), None)
+
+    def search_containers_by_expr_id(self, id, all=False):
+        get_containers = self.containers_info()
+        return filter(lambda c: name_match(id, c["Names"]), get_containers)
+
+    def stop(self, name):
+        if self.get_container(name) is not None:
+            try:
+                containers_url = "http://localhost:4243/containers/%s/stop" % name
```

Comment:
again hostname and port cannot be hard-coded but get from database. We must get the exact hostname where the container was started.